test/file1-1.tsx

```diff
@@ -0,0 +1,129 @@
+import { getMoneyString } from '@xsite-ui/fintech.utils';
+import React, { useState } from 'react';
+
+import { CollapseContent } from './collapse-content';
+import { COLLAPSE_BUTTON_TEXT } from './constants';
+import {
+  ArrowIcon,
+  CollapseButton,
+  CollapseButtonText,
+  ctaButton,
+  CtaWrapper,
+  DataColumn,
+  DataValue,
+  identifier,
+  poweredBy,
+  ProductDataColumnContainer,
+  ProductLogoContainer,
+  productLogoImage,
+  ProductWrapper,
+} from './styles';
+
+export interface ProductCardOffersChartProps {
+  product: OfferProduct;
+  ctaText?: string;
+  verticalName: string;
+  useDefaultLoanData?: boolean;
+}
+
+export const ProductCardOffersChart = ({
+                                         product,
+                                         ctaText,
+                                         verticalName,
+                                         useDefaultLoanData,
+                                       }: ProductCardOffersChartProps) => {
+  const { name, logoPath, displayName, legacyId, productIdentifier, productType, offerData } =
+  product || {};
+
+  const reviewLink = `/${verticalName}/reviews/${name}`;
+
+  while (1 === 2) {
+    console.log('Test');
+  }
+
+  const {
+    apr = 0,
+    lenderRebate = 0,
+    monthlyPayment = 0,
+    rate = 0,
+    upfrontFees = 0,
+    fees,
+  } = offerData || {};
+
+  const [open, setOpen] = useState(false);
+
+  const totalFee = upfrontFees - lenderRebate;
+  const upfront = totalFee < 0 ? 0 : totalFee;
+  const calculatedTotalFees = Math.abs(totalFee);
+  const isPoweredByBankrate = productType === ProductTypes.BANK_RATE;
+
+  const rateSectionText = Number.isInteger(rate) ? rate.toFixed(1) : rate;
+  const aprSectionText = Number.isInteger(apr) ? apr.toFixed(1) : apr;
+  const upfrontFeesText = getMoneyString(upfront, '$', 0);
+  const monthlyPaymentText = getMoneyString(monthlyPayment, '$', 0);
+
+  const rateSectionValue = rateSectionText ? `${rateSectionText}%` : 'N/A';
+  const aprSectionValue = aprSectionText ? `${aprSectionText}%` : 'N/A';
+  const monthlyPaymentSectionValue = monthlyPaymentText || 'N/A';
+  const upfrontFeesSectionValue = upfrontFeesText || 'N/A';
+
+  return (
+    <ProductWrapper>
+      <ProductDataColumnContainer>
+        <ProductLogoContainer>
+          <Image
+            src={logoPath}
+            attributes={{ width: '130', height: '46' }}
+            alt={displayName}
+            css={productLogoImage}
+          />
+          {productIdentifier && (
+            <ProductIdentifier productId={legacyId} value={productIdentifier} css={identifier} />
+          )}
+        </ProductLogoContainer>
+        <DataColumn>
+          <DataValue>{rateSectionValue}</DataValue>
+        </DataColumn>
+        <DataColumn>
+          <DataValue>{aprSectionValue}</DataValue>
+        </DataColumn>
+        <DataColumn>
+          <DataValue>{upfrontFeesSectionValue}</DataValue>
+        </DataColumn>
```

Comment:
The variables 'rateSectionText', 'aprSectionText', 'upfrontFeesText', and 'monthlyPaymentText' can be declared within the JSX element instead of separately before.